From dec61b553a3e44a2d68bcb30fd57cef81f42faeb Mon Sep 17 00:00:00 2001
From: Stephan Gerhold <stephan.gerhold@linaro.org>
Date: Tue, 22 Jul 2025 15:37:19 +0200
Subject: [PATCH 087/263] arm64: dts: qcom: sc8280xp-el2: Add iris
 video-firmware node

Add the video-firmware node with the IOMMU used for loading the iris
firmware to make video acceleration work when running in EL2.

Signed-off-by: Stephan Gerhold <stephan.gerhold@linaro.org>
Signed-off-by: Abel Vesa <abel.vesa@oss.qualcomm.com>
---
 arch/arm64/boot/dts/qcom/sc8280xp-el2.dtso | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sc8280xp-el2.dtso b/arch/arm64/boot/dts/qcom/sc8280xp-el2.dtso
index da2ab9254..e6c60f46c 100644
--- a/arch/arm64/boot/dts/qcom/sc8280xp-el2.dtso
+++ b/arch/arm64/boot/dts/qcom/sc8280xp-el2.dtso
@@ -45,3 +45,9 @@ &pcie_smmu {
 &scm {
 	qcom,shm-bridge-vmid = <QCOM_SCM_VMID_SELF_OWNER>;
 };
+
+&iris {
+	video-firmware {
+		iommus = <&apps_smmu 0x2a02 0x400>;
+	};
+};
-- 
2.43.0

